thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/PluginServiceFactory.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2023 StarTree Inc
+ *
+ * Licensed under the StarTree Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at http://www.startree.ai/legal/startree-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package ai.startree.thirdeye.spi;
+
+public interface PluginServiceFactory<PluginServiceT, ContextT> {
```

Comment:
what is the goal of this?

Reply:
Goal is to provide structure. If you see, all factories have the same behavior. needs a `name()` and a `build()`. Introducing this now. this is used in both the AuthenticatorFactory and the OpenIdConfigProviderFactory. However, this structure will be usable in all factory classes.

Reply:
ok thanks